opencog/truthvalue/AttentionValue.h

```diff
@@ -35,8 +35,10 @@ namespace opencog
  */
 
 //! stores attention in three components: short-term, long-term and very long-term
+//! This class is currently read-only, and assumed to be thread-safe. Beware of
+//! possible race conditions if it is changed
 class AttentionValue;
-typedef std::shared_ptr<AttentionValue> AttentionValuePtr;
+typedef std::shared_ptr<const AttentionValue> AttentionValuePtr;
```

Comment:
But currently, they are not changed. They are replaced instead. As you can see, AttentionValue's public interface is totally const. It has no set* methods. Notice that Agent has setAV/getAV methods to set/get AV values, but no way to change them. In the current state, I think it is nice to mark this class as read-only.